### DISCUSSION TITLE
How to construct an intact polycrystal model from raw data which contains crack

### DISCUSSION THREAD

**Discussion by kun-Jiang** [mtex-toolbox/mtex#2250] [Ask Anything]
Hi there,

I want to construct an intact polycrystal model for finite element modeling. However, there are crack regions in my raw data. How can I reconstruct grains in these regions? I upload a figure showing what I want to do.
![image](https://github.com/user-attachments/assets/aa25acc6-9d16-4f13-95b4-2d9780709a15)

**Comment by ralfHielscher** [comment]
I'm correct with the assumption that you do not simply want to model the grains over the crack but rather make an image transformation that closes the crack?

Ralf.

**Reply by kun-Jiang** [reply]
Yes, I want to close the crack and then simulate the crack propagation by crystal plasticity.

**Comment by ralfHielscher** [comment]
I'm not aware of any publication where a solution of this problem has been described.

**Reply by kun-Jiang** [reply]
As mentioned in the [article](https://doi.org/10.1016/j.jmps.2019.02.012), the crack was closed using image editing software. I'm unsure how they achieved this, so I would like to know if it can be implemented using MTEX.
